daemons/controld/controld_execd.c

```diff
@@ -1988,8 +1988,8 @@ do_lrm_rsc_op(lrm_state_t *lrm_state, lrmd_rsc_info_t *rsc, xmlNode *msg,
         crm_notice("Discarding attempt to perform action %s on %s in state %s "
                    "(shutdown=%s)", operation, rsc->id,
                    fsa_state2string(controld_globals.fsa_state),
-                   pcmk__btoa(pcmk_is_set(controld_globals.fsa_input_register,
-                                          R_SHUTDOWN)));
+                   pcmk__flag_str(controld_globals.fsa_input_register,
+                                  R_SHUTDOWN));
```

Comment:
We have a variety of conventions for "readable string version of something". The `toa` functions are (so far) used for basic types only. `2str`, `2string` `2text` are common but I've been moving away from them. `_str` and `_text` are both used for enumeration values and such, but I've been leaning to `_str` for new stuff. Then there's `readable` which is currently used for things that are only suitable for displaying (in output/logs) and not for processing.

I haven't established any guidelines, but I'm leaning to `toa` for simple types, `readable` for display-only text, and `str` for string equivalents of enum values and other non-basic types. I don't feel strongly about it though so I'm welcome to suggestions.

Reply:
That sounds good to me. `toa` mimics the inverse of standard functions like `atoi()`, so they're terse, but the meaning is pretty clear.

The line might occasionally blur between `readable` and `str` but case-by-case should be fine.

The only downside I can think of for `_str` is that it can look like noun-y or like a variable name (we have some variables that end in `_str`, but clear from context) or equivalently, while `2whatever` is very clear and concise in its intent: convert from one type to another. I don't like overloading a homophone though ("2" for "to"), and "_to_str" is too long.

Reply:
I'm leaning to `_text()` now, to more clearly distinguish from the other uses of `str`

Reply:
I think it reads better too -- `pcmk__flag_text()`, `pcmk__role_text()`, etc.

Reply:
Works for me. FWIW, the "other uses" of `str` can be considered legacy or nonstandard. We normally go with `_s` as the "this variable holds a string" suffix.